### DISCUSSION TITLE
考虑支持预览stl文件嘛

### DISCUSSION THREAD

**Discussion by lyizhuo** [spencerwooo/onedrive-vercel-index#521] [Ideas]
考虑支持预览stl格式的文件嘛

**Comment by Android-KitKat** [comment]
👀 不知道有没有什么好用的 网页3D查看器。
我用过效果最好的是 [Sketchfab](https://sketchfab.com)，但这是个**商用**的。

**Reply by lyizhuo** [reply]
[Babylonjs](https://www.babylonjs.com/)这个呢

**Reply by Android-KitKat** [reply]
好像不错，这个项目有个 Viewer 组件。
[First Model on a Web Page](https://doc.babylonjs.com/start/chap1/first_viewer)
[Babylon.js Viewer](https://doc.babylonjs.com/extensions/babylonViewer)
只需要两行代码，就可以在网页展示 .gltf、.glb、.stl、.obj、.babylon 格式的 3D 模型。(虽然没有 .fbx)
并且默认自带环境，会有基本的灯光、地面、阴影、背景。

甚至还支持动画和进度条以及声音，挺炫酷的。
[BabylonJS Viewer - UFO](https://viewer.babylonjs.com/ufoexample)

**Reply by Android-KitKat** [reply]
我没搞懂怎么在 React 中用 `babylonjs-viewer`。
需要了解的大佬来帮忙，或者指点一下。
@spencerwooo @myl7 

我觉得这个挺实用的。
毕竟 [Sketchfab](https://sketchfab.com) 虽然效果很好，但免费用户每月只能上传 10 个模型。
而如果支持在线查看 3D 模型的话，创作者就能很方便且不受限制的展示模型。

**Comment by spencerwooo** [comment]
- https://github.com/brianzinn/react-babylonjs
- https://github.com/Flux159/next-babylonjs

有时间之后支持上，近期很忙。

**Reply by Android-KitKat** [reply]
好，麻烦大佬了。

我还找到一个，现成能在 React 里用的。
这个支持的文件格式更多。
先贴个链接，到时候可以参考、对比一下。
* [React Three Fiber](https://docs.pmnd.rs/react-three-fiber)
* [drei](https://github.com/pmndrs/drei)
